avm/ptn/azd/aks/main.bicep

```
@@ -0,0 +1,465 @@
+@description('Optional. Specifies the location of AKS cluster. It picks up Resource Group\'s location by default.')
+param location string = resourceGroup().location
+
+@description('Optional. Custom tags to apply to the AKS resources.')
+param tags object = {}
+
+@description('Optional. Network plugin used for building the Kubernetes network.')
+@allowed(['azure', 'kubenet'])
+param networkPlugin string = 'azure'
+
+@description('Optional. Specifies the network policy used for building Kubernetes network. - calico or azure.')
+@allowed(['azure', 'calico'])
+param networkPolicy string = 'azure'
+
+@description('Optional. Specifies the DNS prefix specified when creating the managed cluster.')
+param dnsPrefix string = ''
+
+@description('Optional. The name of the resource group for the managed resources of the AKS cluster.')
+param nodeResourceGroupName string = ''
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Required. Id of the user or app to assign application roles.')
+param principalId string
+
+@description('Optional. The type of principal to assign application roles.')
+@allowed(['Device', 'ForeignGroup', 'Group', 'ServicePrincipal', 'User'])
+param principalType string = 'User'
+
+@description('Optional. Kubernetes Version.')
+param kubernetesVersion string = '1.29'
+
+@description('Optional. Tier of a managed cluster SKU.')
+@allowed([
+  'Free'
+  'Premium'
+  'Standard'
+])
+param skuTier string = 'Standard'
+
+@description('Optional. Network dataplane used in the Kubernetes cluster. Not compatible with kubenet network plugin.')
+@allowed([
+  'azure'
+  'cilium'
+])
+param networkDataplane string?
+
+@description('Optional. Network plugin mode used for building the Kubernetes network. Not compatible with kubenet network plugin.')
+@allowed([
+  'overlay'
+])
+param networkPluginMode string?
+
+@description('Optional. Specifies the CIDR notation IP range from which to assign pod IPs when kubenet is used.')
+param podCidr string?
+
+@description('Optional. A CIDR notation IP range from which to assign service cluster IPs. It must not overlap with any Subnet IP ranges.')
+param serviceCidr string?
+
+@description('Optional. Specifies the IP address assigned to the Kubernetes DNS service. It must be within the Kubernetes service address range specified in serviceCidr.')
+param dnsServiceIP string?
+
+@description('Optional. Specifies the SSH RSA public key string for the Linux nodes.')
+param sshPublicKey string?
+
+@description('Optional. Specifies whether to enable Azure RBAC for Kubernetes authorization.')
+param aadProfileEnableAzureRBAC bool = false
+
+@description('Conditional. Specifies the resource ID of connected application gateway. Required if `ingressApplicationGatewayEnabled` is set to `true`.')
+param appGatewayResourceId string?
+
+@description('Optional. Resource ID of the monitoring log analytics workspace.')
+param monitoringWorkspaceId string?
```

Must be changed to 
```suggestion
param monitoringWorkspaceResourceId string?
```

if it's actually the resource and not workspaceID. This needs to be fixed in other places too